<!DOCTYPE html>
<html>
  <head>
    <title>Deep learning in Biology</title>
    <meta charset="utf-8">
    <meta name="author" content="Karla Fejfarová" />
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link href="libs/remark-css/default-fonts.css" rel="stylesheet" />
    <link rel="stylesheet" href="pres.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Deep learning in Biology
## Journal club
### Karla Fejfarová
### 2018/03/07

---

layout: true

&lt;div class="my-footer"&gt;&lt;span&gt;Journal club: 7-Mar-2018, Karla&lt;/span&gt;&lt;/div&gt;

---

# Papers

LeCun, Y., Bengio, Y. &amp; Hinton, G.  
**Deep learning.**  
*Nature* 521, 436–444 (2015)  
[doi:10.1038/nature14539](https://doi.org/10.1038/nature14539)

Angermueller, C., Pärnamaa, P., Parts, L. &amp; Stegle, O.  
**Deep learning for computational biology.**  
*Mol.Syst Biol* 12:878 (2016)  
[doi:10.15252/msb.20156651](https://doi.org/10.15252/msb.20156651)

Ching et al. (2018)  
**Opportunities And Obstacles For Deep Learning In Biology And Medicine**   
[preprint doi:10.1101/142760](https://doi.org/10.1101/142760)

---

# What is Deep learning?

![](./journal_club_201803_files/first.png)

---

# What is Deep learning?

.pull-left[
Input `\(\longrightarrow\)` hidden layers `\(\longrightarrow\)` output

![](./journal_club_201803_files/def.png)

**Input** - measured variables:
* intensity of a pixel in an image
* expression level of a gene

**Output: **
* labels/categories (supervised learning)
* patterns (unsupervised learning)
]

.pull-right[
Neural networks
![](./journal_club_201803_files/dnn.png)
]
---

# What is Deep learning?

.pull-left[
![](journal_club_201803_files/figure-html/lm1-1.png)&lt;!-- --&gt;
]


---

# What is Deep learning?

.pull-left[
![](journal_club_201803_files/figure-html/lm2-1.png)&lt;!-- --&gt;
]

---

# What is Deep learning?

.pull-left[
![](journal_club_201803_files/figure-html/relu-1.png)&lt;!-- --&gt;
]

--

.pull-right[
![](./journal_club_201803_files/1neuron.png)

]

---

# What is Deep learning?

.pull-left[
![](journal_club_201803_files/figure-html/relu2-1.png)&lt;!-- --&gt;
]

.pull-right[
![](./journal_club_201803_files/neural_net.png)

]

---
background-image: url(./journal_club_201803_files/neuralnetworks.png)
background-size: 80%
background-position: 30% 20%
class: bottom

Image credit: [Neural network Zoo](http://www.asimovinstitute.org/neural-network-zoo/)

---

# What is Deep learning?

.pull-left[
Input `\(\longrightarrow\)` hidden layers `\(\longrightarrow\)` output

![](./journal_club_201803_files/def.png)

**Input** - measured variables:
* intensity of a pixel in an image
* expression level of a gene

**Output: **
* labels/categories (supervised learning)
* patterns (unsupervised learning)

.small[Image from Zeiler &amp; Fergus (2014). [doi: 10.1007/978-3-319-10590-1_53](https://doi.org/10.1007/978-3-319-10590-1_53)]
]

.pull-right[
![](./journal_club_201803_files/vis.png)

]

---

# Why deep learning?

![](./journal_club_201803_files/data.png)
.small[Image credit: [https://hackernoon.com/](https://hackernoon.com/%EF%B8%8F-big-challenge-in-deep-learning-training-data-31a88b97b282)]

---

# The main challenge: get the training data

![](./journal_club_201803_files/dataset.gif)

.small[Image credit: [https://hackernoon.com/](https://hackernoon.com/%EF%B8%8F-big-challenge-in-deep-learning-training-data-31a88b97b282)]
---

# The main challenge: get the training data

.pull-left[
* correct labels

![](./journal_club_201803_files/fibrosis-labels-1.png)
.small[Image credit: [https://lukeoakdenrayner.wordpress.com/2017/12/18/the-chestxray14-dataset-problems/](https://lukeoakdenrayner.wordpress.com/2017/12/18/the-chestxray14-dataset-problems/)]
]
--
.pull-right[
* bias in the data

![](./journal_club_201803_files/bias.png)
.small[[https://medium.com/@yonatanzunger/](https://medium.com/@yonatanzunger/asking-the-right-questions-about-ai-7ed2d9820c48)]

]

---

# Applications

*  find low dimensional representation of cancer gene expression data, discovery of clinically relevant tumour subtypes  
.medium[(Young *et al.* (2017) [doi: 10.1186/s12859-017-1798-2](https://doi.org/10.1186/s12859-017-1798-2) )]
* prediction of DNA methylation sites  
.medium[(Angermueller *et al.* (2017) [doi: 10.1186/s13059-017-1189-z](https://doi.org/10.1186/s13059-017-1189-z))]
* prediction of functional output of DNA sequences  
.medium[(Kelley *et al.* (2016) [doi: 10.1101/gr.200535.115](https://doi.org/10.1101/gr.200535.115))]
* image segmentation (e.g. select liver area in CT images)  
.medium[(Ronneberger *et al.* (2015) [doi: 10.1007/978-3-319-24574-4_28](https://doi.org/10.1007/978-3-319-24574-4_28), ...)]
* prediction of protein function from sequence  
.medium[(Kulmanov *et al.* (2018) [doi: 10.1093/bioinformatics/btx624](https://doi.org/10.1093/bioinformatics/btx624))]
* drug design: generate new compounds predicted to be active against a biological target  
.medium[(Olivecrona *et al.* (2017) [doi: 10.1186/s13321-017-0235-x](https://doi.org/10.1186/s13321-017-0235-x))]
* classification of skin cancer type based on photos  
.medium[(Esteva *et al.* (2017) [doi: 10.1038/nature21056](https://doi.org/10.1038/nature21056))]
* classification of cells according to their cell cycle stage based on image data  
.medium[(Eulenberg *et al.* (2017) [doi: 10.1038/s41467-017-00623-3](https://doi.org/10.1038/s41467-017-00623-3))]
---

# Image Segmentation

Ronneberger O., Fischer P., Brox T. (2015)  
**U-Net: Convolutional Networks for Biomedical Image Segmentation.**  
In: Navab N., Hornegger J., Wells W., Frangi A. (eds) Medical Image Computing and Computer-Assisted Intervention – MICCAI 2015. Lecture Notes in Computer Science, vol 9351. Springer, Cham  
(doi: [10.1007/978-3-319-24574-4_28](https://doi.org/10.1007/978-3-319-24574-4_28))

![](./journal_club_201803_files/unet.png)

---

# Image Segmentation

Christ P. F., Elshaer M. E. A., Ettlinger F. *et al.* (2016)  
**Automatic liver and lesion segmentation in ct using cascaded fully convolutional neural networks and 3D conditional random fields.**    
In: International conference on medical image computing and computer-assisted intervention. Springer International Publishing; 2016. p. 415–23. 4.
(doi: [10.1007/978-3-319-46723-8_48](https://doi.org/10.1007/978-3-319-46723-8_48))
.img_small[
![](./journal_club_201803_files/liver.png)
]

---

# Metabolomics

Fadhl M. Alakwaa, Kumardeep Chaudhary, and Lana X. Garmire (2018)
**Deep Learning Accurately Predicts Estrogen Receptor Status in Breast Cancer Metabolomics Data**
Journal of Proteome Research 2018 17 (1), 337-347
[DOI: 10.1021/acs.jproteome.7b00595](https://doi.org/10.1021/acs.jproteome.7b00595)

.pull-left[
![](./journal_club_201803_files/metab.jpeg)
]

.pull-right[
data: 
* 271 breast cancer samples
* 162 metabolites with known structure (GC-TOFMS)

![](./journal_club_201803_files/metab2.jpg)
]

---

.pull-left[
## Take home message

### The most important thing in deep learning is the **data** used to train the neural network.

![](./journal_club_201803_files/dataend.png)
.small[Image from [hackernoon.com](https://hackernoon.com/%EF%B8%8F-big-challenge-in-deep-learning-training-data-31a88b97b282)]
]

--
.pull-right[
![](./journal_club_201803_files/owl.gif)
## Thank you!
]
    </textarea>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function() {
  var d = document, s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})();</script>

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre']
  }
});
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
